File: EduPortal.API/EduPortal.Dto/Models/SearchResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EduPortal.Dto.Models
{
    public class SearchResult
    {
        public ChildDetails Child { get; set; }
        public List<Graduate>? GraduateMatch { get; set; }
        public List<YeshivaStudent>? YeshivaStudentMatch { get; set; }
    }
}

File: EduPortal-client/src/app/Models/SearchResult.ts
import { ChildDetails } from './ChildDetails';
import { Graduate } from './Graduate';
import { YeshivaStudent } from './YeshivaStudent';

export interface SearchResult {
  child?: ChildDetails;
  graduateMatch?: Graduate[];
  yeshivaStudentMatch?: YeshivaStudent[];
}
